### ISSUE TITLE
Finalize structure of the repo

### ISSUE THREAD

**Issue by gcroci2** [DeepRank/3D-Vac#155]
🚨 TODOs to make the repo public 🚨
- Go over the PRs and close/merge them
   - [x] @gcroci2 
   - [x] @DarioMarzella 
   - [x] @DanLep97  
- Go through folder in src/ and uniform them:
   - [x] folder 3_ and 4_ need to have two subfolders, one for deeprank and one for deeprankcore @DarioMarzella, @DanLep97 move the scripts accordingly
   - check exploration/ and tools/ and create deeprank/deeprankcore subfolders and move there the relevant scripts (plotting results, loss curves, ...)
       - [x] @gcroci2 
       - [x] @DarioMarzella 
       - [x] @DanLep97 
   - put into exploration/ and tools/ all scripts that can be considered as such, outside deeprank/deeprankcore subfolders if they are independent from them (or if they use both packages, e.g., plotting comparisons of deeprank/deeprankcore models)
       - [x] @gcroci2 
       - [x] @DarioMarzella 
       - [x] @DanLep97 
   - move into exploration/ and tools/ scripts that are independent of the package used that now are in other places
       - [x] @gcroci2 
       - [x] @DarioMarzella 
       - [x] @DanLep97 
   - add the readmes in each subfolder
       - [x] @gcroci2 
       - [x] @DarioMarzella 
       - [x] @DanLep97 
- [x] change deeprankcore to DeepRank2 @gcroci2, to be aligned with DeepRank nomenclature and avoiding using the same name as the packages' ones (deeprank/deeprank2)  
- [x] In src/ make the numbering consistent (all needs to start from 0 or 1, folders-wise and python files-wise) @gcroci2 
- [x] create two branches, one for the paper (removing stuff not relevant for it), another one for keeping all the experiments/attempts/scripts: https://github.com/DeepRank/3D-Vac/tree/paper this is the branch for the paper
- [x] update the readme according to the current folders and scripts structure @gcroci2 and adjust the subfolders
- [x] Add MHCFlurry ref
- [x] clean `exploration/' folder. A short description of Dario's notebooks/scripts is still missing.
- [x] Clean the tools/ folder and document eventual remaining scripts. I have never used it, no script is documented so I don't know what I can delete. 
- [ ] update the readme with the links to the data, pre-trained models and outputs (TBD place-holders) 
- [x] Make the repo public (or a fork of the repo)

TODOs asap, but fine also after having made the repo public:
- [x] we mention the additional features in DB4-DeepRank (e.g., desolvation energy). Did we use them? If not, update readme and delete the files accordingly.
- [x] Add Tadija's instructions about EGNN when he sends them.
- [x] Add Tadija's SSL part when he sends it.
